packages/replay/src/replay.ts

```diff
@@ -981,7 +993,15 @@ export class ReplayContainer implements ReplayContainerInterface {
     const envelope = createReplayEnvelope(replayEvent, recordingData, dsn, client.getOptions().tunnel);
 
     try {
-      return await transport.send(envelope);
+      const response = await transport.send(envelope);
+      // TODO (v8): we can remove this guard once transport.send's type signature doesn't include void anymore
+      if (response) {
+        this._rateLimits = updateRateLimits(this._rateLimits, response);
```

Comment:
Hmm, what's the reason to keep this on the class instance? Could we not just pass this through, e.g.:

```js
const rateLimits = updateRateLimits({}, response);
// ...
this._handleRateLimits(rateLimits);
```

?

Reply:
I added `_rateLimits` to the class as I was initially worried about multiple pending requests which would both return different rate limit responses. Also, we keep the rateLimits object alive across multiple requests in the base transport. 

But I guess the risk here is minimal, so I'm happy to change it to a local variable.

Reply:
Yeah, I am also not 100% sure. I guess if there are multiple pending requests, they would just both result in the `_handleRateLimits` being called, right? Maybe we should add some logic there to handle this. Thinking of either:

a) Check if `isPaused() === true`, and if so, do nothing
b) Cancel the previous timeout, if it exists, and create a new one

I'd say a) is the easier one - and it should still work if we run into the rate limit again, as it will just re-pause it then. WDYT?

Reply:
> if there are multiple pending requests, they would just both result in the _handleRateLimits being called, right?

Good point!

>I'd say a) is the easier one

Agreed, will do this!